src/internal/mod.rs

```diff
@@ -0,0 +1,12 @@
+/// Internal trait to get access to the container storing the annotations.
+/// Used for the generic implementation of `AnnotationExt`
+pub trait HasAnnotations {
+    fn annotations(&mut self) -> &mut Vec<String>;
+}
+
+/// Internal trait to get access to the container storing the inner and outer annotations.
+/// Used for the generic implementation of `InnerAndOuterAnnotationExt`
+pub trait HasInnerAndOuterAnnotations {
```

Comment:
I don't know the answer here, but do you think we should rather have `OuterAnnotations`, `InnerAnnotations` and have `InnerAndOuterAnnotations: OuterAnnotations + InnerAnnotations` ?

Perhaps that's a bit "too much", but seems having `InnerAndOuterAnnotations` makes `Annotations` a bit unclear where the annotations are located in that context.

Reply:
It really shouldn't matter since no library user sees them but let me try locally later today

Reply:
Fair point. After you get a feel for it locally, I'll let you make the call then. :+1:

Reply:
Trying it locally showed that it would only work the other way around. `impl <T: InnerAndOuterAnnotations> OuterAnnotations for T` Because `InnerAndOuterAnnotations: OuterAnnotations + InnerAnnotations` would need a item to implement both `OuterAnnotations` and `InnerAnnotations` to be viable to allow implementing `InnerAndOuterAnnotations`.
It would only makes sense if we had a type where we have only inner/outer annotations where the normal name annotations doesn't makes sense?